raiden/blockchain/events.py

```diff
@@ -245,10 +249,40 @@ class BlockchainEvents:
     def __init__(self, chain_id: ChainID):
         self.chain_id = chain_id
         self.event_listeners: List[EventListener] = list()
+        self.last_sync_log_time: Optional[datetime] = None
+        self.last_sync_log_block: BlockNumber = BlockNumber(0)
+
+    def _log_sync_progress(self, to_block: BlockNumber) -> None:
+        """
+        In case we have fallen far behind with synchronizing blockchain events,
+        display a sync progress message every few seconds.
+        """
+        from_block = min(listener.filter.from_block_number() for listener in self.event_listeners)
+        blocks_to_sync = to_block - from_block
+
+        if blocks_to_sync > 100:
+            if self.last_sync_log_time is None:
+                self.last_sync_log_time = datetime.now()
+                self.last_sync_log_block = from_block
+                log.info("Synchronizing blockchain events", blocks_left=blocks_to_sync)
+            else:
+                now = datetime.now()
+                elapsed = (now - self.last_sync_log_time).total_seconds()
+                if elapsed >= 5.0:
+                    blocks_per_second = (from_block - self.last_sync_log_block) / elapsed
+                    log.info(
+                        "Synchronizing blockchain events",
+                        blocks_left=blocks_to_sync,
+                        blocks_per_second=blocks_per_second,
+                    )
+                    self.last_sync_log_time = now
+                    self.last_sync_log_block = from_block
 
     def poll_blockchain_events(self, block_number: BlockNumber) -> Iterable[DecodedEvent]:
         """ Poll for new blockchain events up to `block_number`. """
 
+        self._log_sync_progress(block_number)
```

Comment:
Will this actually be called frequently enough to provide the intended log output? From just looking at this part of the code, it looks like we only call this once and then get stuck in `get_new_entries` for a long time when we have much to update. Is this actually the case or is this batched somehow?